code/_onclick/item_attack.dm

```diff
@@ -1,12 +1,28 @@
 
+/obj/item/proc/melee_attack_chain(mob/user, atom/target, params)
+	if(tool_attack_chain(user, target))
+		return
+	// Return 1 in attackby() to prevent afterattack() effects (when safely moving items for example)
+	var/resolved = target.attackby(src, user, params)
+	if(!resolved && target && !QDELETED(src))
```

Comment:
```suggestion
	if(resolved || QDELETED(target) || QDELETED(src))
		return
	afterattack(target, user, TRUE, params) // TRUE: clicking something Adjacent
```

Reply:
are you sure about the QDELETED(target) ?

Reply:
Yep, sure!